config/test/network-policy/allow-egress-dns.yaml

```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-egress-to-dns
+  namespace: kubemacpool-system
```

Comment:
just to keep it aligned with the others: use `namespace: system`, kustomize adds the prefix later anyways.
Same for the other yamls you added